lib/internal/loader/ModuleJob.js

```diff
@@ -27,26 +24,26 @@ class ModuleJob {
 
     // Wait for the ModuleWrap instance being linked with all dependencies.
     const link = async () => {
-      const dependencyJobs = [];
       ({ module: this.module,
          reflect: this.reflect } = await this.modulePromise);
       if (inspectBrk) {
         const initWrapper = process.binding('inspector').callAndPauseOnStart;
         initWrapper(this.module.instantiate, this.module);
       }
       assert(this.module instanceof ModuleWrap);
-      this.module.link(async (dependencySpecifier) => {
-        const dependencyJobPromise =
-            this.loader.getModuleJob(dependencySpecifier, url);
-        dependencyJobs.push(dependencyJobPromise);
-        const dependencyJob = await dependencyJobPromise;
-        return (await dependencyJob.modulePromise).module;
+
+      const jobs = [];
+      const promises = this.module.link(async (specifier) => {
+        const job = await this.loader.getModuleJob(specifier, url);
+        jobs.push(job);
```

Comment:
If we were to keep a `jobs` array, we would want to store the promises from `getModuleJob` to ensure determinism in the order of `jobs`.